otx/algorithms/detection/tasks/task.py

```diff
@@ -0,0 +1,608 @@
+                predicted_scored_labels += shape.get_labels()
+
+            add_saliency_maps_to_dataset_item(
+                dataset_item=dataset_item,
+                saliency_map=saliency_map,
+                model=self._task_environment.model,
+                labels=labels,
+                predicted_scored_labels=predicted_scored_labels,
+                explain_predicted_classes=explain_predicted_classes,
+                process_saliency_maps=process_saliency_maps,
+            )
+
+    @staticmethod
+    def _generate_training_metrics(learning_curves, scores) -> Iterable[MetricsGroup[Any, Any]]:
+        """Get Training metrics (epochs & scores).
+
+        Parses the mmdetection logs to get metrics from the latest training run
+        :return output List[MetricsGroup]
+        """
+        output: List[MetricsGroup] = []
+
+        # Learning curves.
+        for key, curve in learning_curves.items():
+            len_x, len_y = len(curve.x), len(curve.y)
+            if len_x != len_y:
+                logger.warning(f"Learning curve {key} has inconsistent number of coordinates ({len_x} vs {len_y}.")
+                len_x = min(len_x, len_y)
+                curve.x = curve.x[:len_x]
+                curve.y = curve.y[:len_x]
+            metric_curve = CurveMetric(
+                xs=np.nan_to_num(curve.x).tolist(),
+                ys=np.nan_to_num(curve.y).tolist(),
+                name=key,
+            )
+            visualization_info = LineChartInfo(name=key, x_axis_label="Epoch", y_axis_label=key)
+            output.append(LineMetricsGroup(metrics=[metric_curve], visualization_info=visualization_info))
+
+        # Final mAP value on the validation set.
+        output.append(
+            BarMetricsGroup(
+                metrics=[ScoreMetric(value=scores, name="mAP")],
+                visualization_info=BarChartInfo("Validation score", visualization_type=VisualizationType.RADIAL_BAR),
+            )
+        )
+
+        return output
+
+    def save_model(self, output_model: ModelEntity):
+        """Save best model weights in DetectionTrainTask."""
+        logger.info("called save_model")
+        buffer = io.BytesIO()
+        hyperparams_str = ids_to_strings(cfg_helper.convert(self._hyperparams, dict, enum_to_str=True))
+        labels = {label.name: label.color.rgb_tuple for label in self._labels}
+        model_ckpt = torch.load(self._model_ckpt)
+        modelinfo = {
+            "model": model_ckpt,
+            "config": hyperparams_str,
+            "labels": labels,
+            "confidence_threshold": self.confidence_threshold,
+            "VERSION": 1,
+        }
+
+        torch.save(modelinfo, buffer)
+        output_model.set_data("weights.pth", buffer.getvalue())
+        output_model.set_data(
+            "label_schema.json",
+            label_schema_to_bytes(self._task_environment.label_schema),
+        )
+        output_model.precision = self._precision
+
+    def cancel_training(self):
```

Comment:
Hope we can use other interface to cancel and clean the job. It looks like we need for all other tasks.

Reply:
Yes, as I wrote in the explanation material, I merge OTXBaseTask and OTXDetection task for review to see less classes.
We will split OTXBaseTask from current OTXDetectionTask